src/TextInput/index.js

```diff
@@ -104,7 +104,7 @@ const TextInput = React.forwardRef((props, forwardedRef) => {
           onBlur={_onBlur}
           ref={forwardedRef}
           type={type}
-          required
+          required={props.required}
```

Comment:
technically we can already override this via spread props, but I like the more explicit prop here